addons/stock/models/stock_orderpoint.py

```diff
@@ -109,6 +112,42 @@ def _compute_show_supply_warning(self):
         for orderpoint in self:
             orderpoint.show_supply_warning = not orderpoint.rule_ids
 
+    @api.depends('location_id', 'product_min_qty', 'route_id', 'product_id.route_ids', 'product_id.stock_move_ids',
+                 'product_id.stock_move_ids.state', 'product_id.seller_ids', 'product_id.seller_ids.delay', 'company_id.horizon_days')
+    def _compute_deadline_date(self):
+        """ This function checks retrieves all the outgoing moves and incoming until the lead_horizon_date
+        and adds them to the qty_on_hand. The first instance when the qty_on_hand dips below the
+        product_min_qty is the deadline date. """
+        for orderpoint in self:
+            if orderpoint.qty_on_hand < orderpoint.product_min_qty:
+                orderpoint.deadline_date = fields.Date.today()
+            else:
+                _, domain_move_in, domain_move_out = orderpoint.product_id._get_domain_locations()
+                domain_move_in = ([('product_id', '=', orderpoint.product_id.id)]
+                                  + [('state', 'in', ('waiting', 'confirmed', 'assigned', 'partially_available'))]
+                                  + domain_move_in
+                                  + [('date', '<=', orderpoint.lead_horizon_date)])
+                domain_move_out = ([('product_id', '=', orderpoint.product_id.id)]
+                                  + [('state', 'in', ('waiting', 'confirmed', 'assigned', 'partially_available'))]
+                                  + domain_move_out
+                                  + [('date', '<=', orderpoint.lead_horizon_date)])
+
+                Move = self.env['stock.move'].with_context(active_test=False)
+                moves_dict = defaultdict(float)
+                for in_date, in_qty in Move._read_group(domain_move_in, ['date:day'], ['product_qty:sum']):
+                    moves_dict[in_date.date()] += in_qty
+                for out_date, out_qty in Move._read_group(domain_move_out, ['date:day'], ['product_qty:sum']):
```

Comment:
these might also have to be batched to gather all the info for all orderpoints at once rather than doing 2 `_read_group`s per orderpoint